libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -161,9 +161,15 @@ int lua_mavlink_send_chan(lua_State *L) {
     const int arg_offset = (luaL_testudata(L, 1, "mavlink") != NULL) ? 1 : 0;
 
     binding_argcheck(L, 3+arg_offset);
-    
+
     const mavlink_channel_t chan = (mavlink_channel_t)get_uint32(L, 1+arg_offset, 0, MAVLINK_COMM_NUM_BUFFERS - 1);
 
+    // Check if the channel is valid
+    if (chan >= gcs().num_gcs()) {
+        lua_pushboolean(L, false);
```

Comment:
There might be a argument to return `nil` here rather than false so the script writer can tell the difference between a failed send and the channel not existing. 

We could also make it a error of course, but then we would have to add a binding for the script writer to get `gcs().num_gcs()`.

Reply:
Done.